translate/cloud-client/src/main/java/com/example/translate/BatchTranslateTextWithModel.java

```diff
@@ -0,0 +1,104 @@
+
+public class BatchTranslateTextWithModel {
+
+  public static void batchTranslateTextWithModel()
+      throws InterruptedException, ExecutionException, IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "YOUR-PROJECT-ID";
+    // Supported Languages: https://cloud.google.com/translate/docs/languages
+    String sourceLanguage = "your-source-language";
+    String targetLanguage = "your-target-language";
+    String inputUri = "gs://your-gcs-bucket/path/to/input/file.txt";
+    String outputUri = "gs://your-gcs-bucket/path/to/results/";
+    String modelId = "YOUR-MODEL-ID";
+    batchTranslateTextWithModel(
+        projectId, sourceLanguage, targetLanguage, inputUri, outputUri, modelId);
+  }
+
+  // Batch translate text using AutoML Translation model
+  public static void batchTranslateTextWithModel(
+      String projectId,
+      String sourceLanguage,
+      String targetLanguage,
+      String inputUri,
+      String outputUri,
+      String modelId)
+      throws IOException, ExecutionException, InterruptedException {
+
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (TranslationServiceClient client = TranslationServiceClient.create()) {
+      // Supported Locations: `global`, [glossary location], or [model location]
+      // Glossaries must be hosted in `us-central1`
+      // Custom Models must use the same location as your model. (us-central1)
+      String location = "us-central1";
+      LocationName parent = LocationName.of(projectId, location);
+
+      GcsSource gcsSource = GcsSource.newBuilder().setInputUri(inputUri).build();
+      // Supported Mime Types: https://cloud.google.com/translate/docs/supported-formats
+      InputConfig inputConfig =
+          InputConfig.newBuilder().setGcsSource(gcsSource).setMimeType("text/plain").build();
+
+      GcsDestination gcsDestination =
+          GcsDestination.newBuilder().setOutputUriPrefix(outputUri).build();
+      OutputConfig outputConfig =
+          OutputConfig.newBuilder().setGcsDestination(gcsDestination).build();
+
+      String modelPath =
+          String.format("projects/%s/locations/%s/models/%s", projectId, location, modelId);
+
+      BatchTranslateTextRequest request =
+          BatchTranslateTextRequest.newBuilder()
+              .setParent(parent.toString())
+              .setSourceLanguageCode(sourceLanguage)
+              .addTargetLanguageCodes(targetLanguage)
+              .addInputConfigs(inputConfig)
+              .setOutputConfig(outputConfig)
+              .putModels(targetLanguage, modelPath)
+              .build();
+
+      OperationFuture<BatchTranslateResponse, BatchTranslateMetadata> future =
```

Comment:
nit: A comment here clarifying that this will send the request to the server to start the job asynchronously would also be useful.